tensorflow/python/compiler/tensorrt/trt_convert.py

```diff
@@ -271,11 +298,14 @@ def _get_tensorrt_rewriter_config(conversion_params,
     # need to run constant folding again.
     rewriter_config_with_trt.optimizers.extend(
         ["constfold", "layout", "constfold"])
+
   rewriter_config_with_trt.meta_optimizer_iterations = (
       rewriter_config_pb2.RewriterConfig.ONE)
   optimizer = rewriter_config_with_trt.custom_optimizers.add()
-  # Add a constfold optimizer to cleanup the unused Const nodes.
-  rewriter_config_with_trt.custom_optimizers.add().name = "constfold"
+
+  if not disable_non_trt_optimizers:
```

Comment:
To fix the bugs mentioned here, I need to fix all of them. For the Cast OP. I suppose it will be a mix of all of them since the bug only occurs in Offline mode & INT8 Calibration.

> This PR fixes the following cases:
> 
> * Convert Offline wasn't properly deactivating Non TRT Grappler Opt:
>   
>   * https://github.com/tensorflow/tensorflow/pull/45620/files#diff-39ad5be340981b91107e595a47a42caf7da5a8436448d64a8914c4696384d5a2R336
>   * https://github.com/tensorflow/tensorflow/pull/45620/files#diff-8e470ff4b5b82fda4fb3fe008b694323d63484f3cbf490d4e3ba104894f914e6R676-R681
> * During GraphState !=  Inference, same issue:
>   https://github.com/tensorflow/tensorflow/pull/45620/files#diff-39ad5be340981b91107e595a47a42caf7da5a8436448d64a8914c4696384d5a2R336
> * A constfold was systematically added even when non TRT Grappler Opt were deactivated: https://github.com/tensorflow/tensorflow/pull/45620/files#diff-8e470ff4b5b82fda4fb3fe008b694323d63484f3cbf490d4e3ba104894f914e6R306-R308
> 
> We also forgot to force deactivate:
> 
> * `auto_mixed_precision=off`: https://github.com/tensorflow/tensorflow/pull/45620/files#diff-8e470ff4b5b82fda4fb3fe008b694323d63484f3cbf490d4e3ba104894f914e6R237
> * `min_graph_nodes=-1`: https://github.com/tensorflow/tensorflow/pull/45620/files#diff-8e470ff4b5b82fda4fb3fe008b694323d63484f3cbf490d4e3ba104894f914e6R248
> 
> As a note, I also took the opportunity to sort alphabetically the rewriter_cfg flags that we deactivate. It was difficult to read and find something.